### ISSUE TITLE
`pycloudlib.instance.BaseInstance._wait_for_system` assumes `cloud-init status --wait` will exit zero

### ISSUE THREAD

**Issue by OddBloke** [canonical/pycloudlib#29]
`cloud-init status --wait` will exit non-zero if there were errors running cloud-init. As we are now using `pycloudlib` for cloud-init's integration testing (where we will want to see our test _assertions_ fail while developing them, and may want to launch "broken" instances to test recovery), this should no longer be considered a hard failure.

This is the code in question: https://github.com/canonical/pycloudlib/blob/fcf6268df72540b99872c1312b8716a7e88b7564/pycloudlib/instance.py#L363-L367

(On focal, at least, the `runlevel` fallback doesn't trigger because `$(runlevel)` gives `N 5`.)

**Comment by OddBloke** [comment]
@blackboxsw @lucasmoura Can either of you comment on whether or not the UA client integration tests rely on cloud-init failures (on xenial and later, I suppose) being raised as exceptions at this stage? I don't want to regress tests over there with my change for this.

**Comment by OddBloke** [comment]
@lucasmoura The issue isn't if the tests _rely_ on cloud-init failing, because the existence of this bug means that we can infer that there are no such tests. My question/point is more that if you aren't currently testing that cloud-init succeeded explicitly, then your test coverage will change slightly: if cloud-init fails, tests would previously have failed but will now pass.

(It may be that the success/failure of cloud-init is orthogonal to your testing, of course, in which case this change should be acceptable.)

**Comment by OddBloke** [comment]
Testing with my lower-level fix in #32: we now get `OSError: cloud-init failed to start: out:  error:`.

**Comment by lucasmoura** [comment]
@OddBloke Yes you are right, I missed that regarding your question. Yes, for for cloud images test we explicitly check to see if cloud-init succeeded.

It wouldn't be hard for us to adapt for that case where cloud-init fails to run. However, maybe we should parametrize the `_wait_for_system` call to allow cloud-init errors to not raise that exception. If I am not mistaken all of the cloud instances perform that check by calling the `wait` method. Maybe we can keep the old behavior as is and parametrize those methods to allow cloud-init integration tests to run properly

**Comment by OddBloke** [comment]
I have a fix for this ready locally; it will need #34 to land, then my refactor for #35 to land, then I can propose it.
